### PR TITLE
feat: 🍰 Group Members Management

### LINKED ISSUE #5344: 🚀 [Feature] Add Group Members Management
## :rocket: Feature Request

in the group settings, the following items should function under the member tab.
- list of all members of a group. 
- member role becomes changeable

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pull Request

- member list of a group is displayed
- member role can be changed

### Issues

- fixes #5344 
 
image

### Todo

- [x] test new pages and components
- [x] remove my setting `My Groups`? from settings page
- [x] remove a member?
  - [x] implement later, see issue #5389
- [x] change group type?
  - [x] implement later, see issue #5387
- [x] translate:
  - [x] select options `groupType`
  - [x] select options `group.actionRadius`
- [x] remove locale:
  - [x] `group.unfollowing`?
  - [x] `group.visibility`
  - [x] `group.showMyCreatedGroups`?
  - [x] `group.showAllMyGroups`?
  - [x] `group.back`
- [x] followup issue #5397
  - [x] may have a look it if something should be already in here …
- [x] cleanup: search for `Wolle`

### CODE REVIEW

**Comment by ogerly** [comment]
group button in menu
image

File: webapp/pages/group/edit/_id/members.vue [hunk 1]
```diff
@@ -1,16 +1,46 @@
...
+
 export default {
   components: {
     GroupMember,
   },
+  props: {
+    group: {
+      type: Object,
+      required: false,
+      default: () => ({}),
```
Comment (Tirokk) [inline webapp/pages/group/edit/_id/members.vue:19]:
```suggestion
      default: {},
```

File: webapp/pages/group/edit/_id/members.vue [hunk 2]
```diff
@@ -1,16 +1,46 @@
...
+    async groupMembersQueryList() {
+      try {
+        const response = await this.$apollo.query({
+          query: groupMembersQuery,
+          variables: { id: this.group.id },
+        })
+        this.responseGroupMembersQuery = response.data.GroupMembers
+      } catch (error) {
+        this.responseGroupMembersQuery = []
+      } finally {
```
Comment (Tirokk) [inline webapp/pages/group/edit/_id/members.vue:37]: Toast the error … ?

File: webapp/pages/my-groups.vue [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
     <br />
     <br />
-    <group-card :items="responseGroupListQuery" />
+    <group-list :items="responseGroupListQuery" />
   </div>
 </template>
 <script>
-import GroupTeaser from '~/components/Group/GroupTeaser.vue'
-import GroupCard from '~/components/Group/GroupCard.vue'
+import GroupList from '~/components/Group/GroupList.vue'
```
Comment (Tirokk) [inline webapp/pages/my-groups.vue:26]:
```suggestion
import GroupList from '~/components/Group/GroupList'
```

**Review by Tirokk — DISMISSED** [review]

File: webapp/pages/group/edit/_id/members.vue (hunk 1 above)
Comment (ogerly) [reply webapp/pages/group/edit/_id/members.vue:19]: done

File: webapp/pages/group/create.vue [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
-    <ds-flex :width="{ base: '100%' }" gutter="base">
-      <ds-flex-item :width="{ base: '100%', md: 5 }">
-        <group-form @createGroup="createGroup" />
-      </ds-flex-item>
-      <ds-flex-item :width="{ base: '100%', md: 1 }">&nbsp;</ds-flex-item>
-    </ds-flex>
-    <hr />
-    <group-member />
+    <ds-section>
+      <h1 class="ds-heading ds-heading-h1">{{ $t('group.newGroup') }}Create New Groupe</h1>
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/pages/group/create.vue:4]: done

File: webapp/pages/group/edit/_id/members.vue (hunk 2 above)
Comment (ogerly) [reply webapp/pages/group/edit/_id/members.vue:37]: done

File: webapp/components/Group/GroupForm.vue [hunk 5]
```diff
@@ -142,9 +178,94 @@ export default {
...
+      const place = encodeURIComponent(value)
+      const lang = this.$i18n.locale()
+
+      const {
+        data: { queryLocations: res },
+      } = await this.$apollo.query({ query: queryLocations(), variables: { place, lang } })
+
+      this.cities = this.processLocationsResult(res)
+      this.loadingGeo = false
+    },
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:217]: ok

File: webapp/components/Group/GroupList.vue [hunk 6]
```diff
@@ -0,0 +1,119 @@
+<template>
+  <ds-container class="group-card">
+    <ds-space>
+      <div @click="onlyOwnerGroups(true)" ref="myGruops">
+        <ds-button>{{ $t('group.show-my-created-groups') }}</ds-button>
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupList.vue:5]: done

File: webapp/components/Group/GroupList.vue [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+              </nuxt-link>
+              <ds-text size="small">
+                {{ item.groupType }}
+                <ds-chip v-if="item.myRole === 'owner'" color="inverse">{{ item.myRole }}</ds-chip>
+                <ds-chip v-if="item.myRole === 'usual' || item.myRole === 'pending'">
+                  {{ item.myRole }}
+                </ds-chip>
+              </ds-text>
+              <ds-space margin-top="small">
+                <ds-text bold>{{ item.about }}</ds-text>
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupList.vue:27]: done

File: webapp/components/Group/GroupList.vue [hunk 8]
```diff
@@ -0,0 +1,119 @@
+<template>
+  <ds-container class="group-card">
+    <ds-space>
+      <div @click="onlyOwnerGroups(true)" ref="myGruops">
+        <ds-button>{{ $t('group.show-my-created-groups') }}</ds-button>
+      </div>
+      <div @click="onlyOwnerGroups(false)" ref="allGruops" hidden>
+        <ds-button>{{ $t('group.show-all-my-groups') }}</ds-button>
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupList.vue:8]: done

File: webapp/components/Group/GroupList.vue [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    },
+    async joinGroup(value) {
+      const { id } = value
+      const variables = { groupId: id, userId: this.$store.getters['auth/user'].id }
+      try {
+        await this.$apollo.mutate({
+          mutation: joinGroupMutation,
+          variables,
+        })
+        this.$toast.success(this.$t('group.group-created'))
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupList.vue:82]: done

File: webapp/components/Group/GroupMember.vue [hunk 10]
```diff
@@ -1,56 +1,91 @@
...
+        </template>
+      </ds-table>
+    </base-card>
+    <ds-modal
+      v-if="isOpen"
+      v-model="isOpen"
+      :title="`${$t('group.remove-member')} ?`"
+      force
+      extended
+      :confirm-label="$t('group.remove-member')"
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupMember.vue:40]: done

File: webapp/components/Group/GroupMember.vue [hunk 11]
```diff
@@ -1,56 +1,91 @@
...
-      if (index > -1) {
-        this.tableData.splice(index, 1)
+    async changeMemberRole(id, event) {
+      const newRole = event.target.value
+      try {
+        await this.$apollo.mutate({
+          mutation: changeGroupMemberRoleMutation,
+          variables: { groupId: this.groupId, userId: id, roleInGroup: newRole },
+        })
+        this.$toast.success(this.$t('group.change-member-role', { role: newRole }))
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupMember.vue:77]: done

File: webapp/components/Group/GroupMember.vue [hunk 12]
```diff
@@ -1,56 +1,91 @@
...
+          >
+            delete
+          </ds-button>
+        </template>
+      </ds-table>
+    </base-card>
+    <ds-modal
+      v-if="isOpen"
+      v-model="isOpen"
+      :title="`${$t('group.remove-member')} ?`"
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupMember.vue:37]: done

File: webapp/components/Group/GroupMember.vue (hunk 10 above)
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupMember.vue:40]: done

File: webapp/locales/de.json [hunk 13]
```diff
@@ -383,9 +383,14 @@
       "regional": "Regionale Gruppe"
     },
     "actionRadius": "Aktionsradius",
+    "back": "zurück",
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/locales/de.json:386]: done

File: webapp/locales/en.json [hunk 14]
```diff
@@ -383,9 +383,14 @@
       "regional": "Regional Group"
     },
     "actionRadius": "Action radius",
+    "back": "back",
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/locales/en.json:386]: done

File: webapp/plugins/v-tooltip.js [hunk 15]
```diff
@@ -3,7 +3,7 @@ import VTooltip from 'v-tooltip'
 
 Vue.use(VTooltip, {
   defaultDelay: {
-    show: 1500,
+    show: 750,
```
Comment (Tirokk) [inline webapp/plugins/v-tooltip.js:6]: @ogerly gave the hint that 1500ms seems to be to much. So we go for 750ms now.
Think that finds you agree. @Mogge

File: webapp/pages/group/edit/_id/members.vue [hunk 16]
```diff
@@ -1,16 +1,53 @@
...
   },
+  props: {
+    group: {
+      type: Object,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      GroupMembers: [],
```
Comment (Mogge) [inline webapp/pages/group/edit/_id/members.vue:27]: Is this needed?

**Review by Mogge — APPROVED** [review]
As it works, I approve. I have several suggestions to improve the code. May be we can have them in issues.

File: webapp/components/Group/GroupForm.vue [hunk 17]
```diff
@@ -89,24 +156,33 @@ export default {
...
         name: { required: true, min: NAME_LENGTH_MIN, max: NAME_LENGTH_MAX },
+        slug: { required: false },
         groupType: { required: true },
         about: { required: true },
         description: { required: true },
         actionRadius: { required: true },
+        locationName: { required: false },
         categoryIds: {
           type: 'array',
           required: this.categoriesActive,
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:188]: done

File: webapp/components/Group/GroupForm.vue [hunk 18]
```diff
@@ -123,16 +199,54 @@ export default {
       },
     }
   },
+  computed: {
+    submitDisable() {
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:203]: deleted

File: webapp/components/Group/GroupForm.vue [hunk 19]
```diff
@@ -123,16 +199,54 @@ export default {
...
+        this.formData.name === '' ||
+        this.formData.groupType === '' ||
+        // this.formData.about === '' || // not mandatory
+        this.formData.description === '' ||
+        this.formData.actionRadius === '' ||
+        // this.formData.locationName === '' || // not mandatory
+        this.formData.categoryIds.length === 0
+      )
+    },
+    submitDisableEdit() {
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:214]: deleted

File: webapp/components/Group/GroupForm.vue [hunk 20]
```diff
@@ -123,16 +199,54 @@ export default {
...
+        // this.formData.groupType === this.group.groupType && // can not be changed for now
+        this.formData.about === this.group.about &&
+        this.formData.description === this.group.description &&
+        this.formData.actionRadius === this.group.actionRadius &&
+        this.formData.locationName === (this.group.locationName ? this.group.locationName : '') &&
+        this.sameCategories
+      )
+    },
+    sameCategories() {
+      const formDataCategories = this.formData.categoryIds.map((id) => id).sort()
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:227]: deleted

File: webapp/pages/group/edit/_id/members.vue (hunk 16 above)
Comment (ogerly) [reply webapp/pages/group/edit/_id/members.vue:27]: deleted

File: webapp/components/Group/GroupForm.vue [hunk 21]
```diff
@@ -142,9 +256,64 @@ export default {
...
+      this.formData.groupType = event.target.value
+    },
+    changeActionRadius(event) {
+      this.formData.actionRadius = event.target.value
+    },
+    handleCityInput(value) {
+      clearTimeout(timeout)
+      timeout = setTimeout(() => this.requestGeoData(value), 500)
+    },
+    processLocationsResult(places) {
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:269]: https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/5440

[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:269]: https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/5441

File: webapp/components/Group/GroupForm.vue [hunk 22]
```diff
@@ -1,75 +1,142 @@
...
-          placeholder="Radius ..."
-        ></ds-select>
+          icon="globe"
+          :placeholder="`${$t('group.actionRadius')} …`"
+        ></ds-select> -->
+        <ds-text class="select-label">
+          {{ $t('group.actionRadius') }}
+        </ds-text>
+        <ds-icon class="select-icon" name="globe" />
+        <select
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:80]: https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/5441

File: webapp/components/Group/GroupForm.vue [hunk 23]
```diff
@@ -1,75 +1,142 @@
...
+        :label-prop="'label'"
+        :options="groupTypeOptions"
+        icon="user"
+        :placeholder="$t('group.type') + ' …'"
+      ></ds-select> -->
+      <ds-text class="select-label">
+        {{ $t('group.type') }}
+      </ds-text>
+      <ds-icon class="select-icon" name="user" />
+      <select
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:38]: https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/5441

**Pull request merged**